include/laser_filters/scan_shadows_filter.h

```diff
@@ -67,6 +74,12 @@ class ScanShadowsFilter : public filters::FilterBase<sensor_msgs::LaserScan>
   /**@b Configure the filter from XML */
   bool configure()
   {
+    ros::NodeHandle private_nh("~" + getName());
+    dyn_server_.reset(new dynamic_reconfigure::Server<laser_filters::ScanShadowsFilterConfig>(own_mutex_, private_nh));
+    dynamic_reconfigure::Server<laser_filters::ScanShadowsFilterConfig>::CallbackType f;
+    f = boost::bind(&laser_filters::ScanShadowsFilter::reconfigureCB, this, _1, _2);
+    dyn_server_->setCallback(f);
+
```

Comment:
Are the `getParam()` calls below still needed? Dynamic reconfigure should read its initial values from the params anyway, right? Or did the variable names change?

Reply:
> Are the getParam() calls below still needed? Dynamic reconfigure should read its initial values from the params anyway, right? Or did the variable names change?

If we want backwards compatibility yes since all `filter` instances read their parameters from the `param` section of the filter defined in the chain. Dynamic reconfigure also sets is parameters in a separate namespace (cannot be done in another way imo) but initializes with params defined the chain.

IMO, this is a limitation of the filter chain but I think having these parameters dynamically reconfigurable is a must have features for these kind of filters.

Reply:
Yes, that makes sense - thanks for keeping it backwards compatible!